dandiapi/api/views/dandiset.py

```diff
@@ -408,7 +409,17 @@ def users(self, request, dandiset__pk):
             # All owners found
             owners = user_owners + [acc.user for acc in socialaccount_owners]
             removed_owners, added_owners = dandiset.set_owners(owners)
-            dandiset.save()
+            with transaction.atomic():
+                dandiset.save()
+
+                if removed_owners or added_owners:
+                    audit_record = AuditRecord.change_owners(
+                        dandiset=dandiset,
+                        user=request.user,
+                        removed_owners=removed_owners,
+                        added_owners=added_owners,
+                    )
+                    audit_record.save()
```

Comment:
> I'd like some ideas on what to do. Should we remove .remove_owner()? Should we inline the contents of .add_owner() rather than leave it as a method?

I think both of those suggestions would be welcome. Especially since `add_owner` is only ever called at dandiset creation, the code would reduce to simply:

```python
assign_perm('owner', user, dandiset)
```

Reply:
Also, since you're creating a transaction here, should the call to `dandiset.set_owners` be included in that?

Reply:
> I think both of those suggestions would be welcome.

Removed `add_owner()` and `remove_owner()` in f772fc5481f3b1a02f95da850bd618035f941311.

Reply:
> Also, since you're creating a transaction here, should the call to dandiset.set_owners be included in that?

Good catch. Done in 0337f41c67ff92bcde3bf20ea970328cde7d6124.